opencti-platform/opencti-front/src/private/components/data/ingestionTaxii/IngestionTaxiiCreation.jsx

```diff
@@ -185,62 +193,116 @@ const IngestionTaxiiCreation = (props) => {
                 </MenuItem>
               </Field>
               {values.authentication_type === BASIC_AUTH && (
-                <>
-                  <Field
-                    component={TextField}
-                    variant="standard"
-                    name="username"
-                    label={t('Username')}
-                    fullWidth={true}
-                    style={fieldSpacingContainerStyle}
-                  />
+              <>
+                <Field
+                  component={TextField}
+                  variant="standard"
+                  name="username"
+                  label={t('Username')}
+                  fullWidth={true}
+                  style={fieldSpacingContainerStyle}
+                />
+                <div style={{ position: 'relative', display: 'flex', alignItems: 'center' }}>
                   <Field
                     component={TextField}
                     variant="standard"
                     name="password"
+                    type={showToken ? 'text' : 'password'}
                     label={t('Password')}
                     fullWidth={true}
                     style={fieldSpacingContainerStyle}
                   />
-                </>
+                  <IconButton
+                    onClick={toggleTokenVisibility}
+                    aria-label={showToken ? t('Hide') : t('Show')}
+                    style={{
+                      position: 'absolute',
+                      right: 1,
+                      top: '60%',
+                      margin: 0,
+                      padding: 0,
+                      zIndex: 1,
+                    }}
+                    disableRipple
+                  >
+                    {showToken ? <VisibilityOff /> : <Visibility />}
+                  </IconButton>
+                </div>
+              </>
```

Comment:
I think this could be in a dedicated component, with a prop `onToggle(toggled)`. The component could handle his own `showToken` state.

Reply:
actually, you could also take the Field above also. And add props like "label".